main/java/io/github/azagniotov/stubby4j/yaml/YAMLMerger.java

```diff
@@ -0,0 +1,87 @@
+package io.github.azagniotov.stubby4j.yaml;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.FileSystems;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.PathMatcher;
+import java.util.List;
+
+import static io.github.azagniotov.stubby4j.utils.FileUtils.quietIO;
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static java.nio.file.Files.delete;
+import static java.nio.file.Files.exists;
+import static java.nio.file.Files.isDirectory;
+import static java.nio.file.Files.newBufferedWriter;
+import static java.nio.file.StandardOpenOption.APPEND;
+import static java.nio.file.StandardOpenOption.CREATE;
+import static java.nio.file.StandardOpenOption.WRITE;
+import static java.util.stream.Collectors.toList;
+
+/**
+ * @author Abhijit Sarkar
+ */
+public final class YAMLMerger {
+    private static final Logger LOGGER = LoggerFactory.getLogger(YAMLMerger.class);
+    private static final PathMatcher YAML_MATCHER = FileSystems.getDefault()
+            .getPathMatcher("glob:**.{yaml,yml}");
+
+    private YAMLMerger() {
+    }
+
+    public static File merge(File config) throws IOException {
+        Path configDir = config.toPath();
+        LOGGER.info("Config dir: {}.", configDir.toAbsolutePath());
+
+        if (!exists(configDir) || !isDirectory(configDir)) {
+            throw new IllegalArgumentException(
+                    String.format("%s doesn't exist or is not a directory.", configDir.toAbsolutePath())
+            );
+        }
+
+        List<Path> files = findAllYaml(configDir);
```

Comment:
What is the need to actually write into a merged temp file? Why not to just merge loaded sub-files in memory? Also, did you test the merge functionality when `--watch` command line argument is provided - will the changes to one of the sub files cause a reload?